### PR TITLE
Added JWT token

### PR DESCRIPTION
**Pull request by PatrykJurczyk**
You have to add TOKEN_SECRET = sekretnytoken to file .env

### CODE REVIEW

File: src/controllers/user.controllers.js [hunk 1]
```diff
@@ -37,5 +39,8 @@ export const userLogin = async (data, response) => {
   const validPass = await bcrypt.compare(data.password, user.password);
   if (!validPass) return response.status(StatusCodes.BAD_REQUEST).send('Email or password is wrong');
 
+  const token = jsonwebtoken.sign({_id:  user._id}, process.env.TOKEN_SECRET)
+  response.header('auth-token', token);
+
   response.send(`Wiatm ${user.name}!`);
```
Comment (mgoniszewska) [inline src/controllers/user.controllers.js:45]: Typo :)

File: src/controllers/user.controllers.js [hunk 2]
```diff
@@ -37,5 +39,8 @@ export const userLogin = async (data, response) => {
   const validPass = await bcrypt.compare(data.password, user.password);
   if (!validPass) return response.status(StatusCodes.BAD_REQUEST).send('Email or password is wrong');
 
+  const token = jsonwebtoken.sign({_id:  user._id}, process.env.TOKEN_SECRET)
+  response.header('auth-token', token);
```
Comment (mgoniszewska) [inline src/controllers/user.controllers.js:43]: Shouldn't token be sent in response's body?

Comment (PatrykJurczyk) [reply src/controllers/user.controllers.js:43]: In PR user-logout-endpoint i set thid token to cookies like this "response.cookie('auth', token, { maxAge: 900000, httpOnly: true });"

**Review by mgoniszewska — APPROVED** [review]

File: src/controllers/user.controllers.js [hunk 3]
```diff
@@ -2,6 +2,8 @@ import User from '../models/user';
 import { registerValidation, loginValidation } from '../routes/user/validation';
 import { StatusCodes } from 'http-status-codes';
 import bcrypt from 'bcryptjs';
+import jsonwebtoken from 'jsonwebtoken';
+import res from 'express/lib/response';
```
Comment (ike-tom) [inline src/controllers/user.controllers.js:6]: 'res' is declared but its value is never read.

File: src/controllers/user.controllers.js [hunk 4]
```diff
@@ -37,5 +39,8 @@ export const userLogin = async (data, response) => {
   const validPass = await bcrypt.compare(data.password, user.password);
   if (!validPass) return response.status(StatusCodes.BAD_REQUEST).send('Email or password is wrong');
 
+  const token = jsonwebtoken.sign({_id:  user._id}, process.env.TOKEN_SECRET)
```
Comment (ike-tom) [inline src/controllers/user.controllers.js:42]: should we add TOKEN_SECRET to the env example(but without any value)?

File: src/routes/user/verifyToken.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { StatusCodes } from 'http-status-codes';
+import jsonwebtoken from 'jsonwebtoken';
+
+const auth = (req, res, next) => {
+  const token = req.header('auth-token');
+  if (!token) return res.status(StatusCodes.BAD_REQUEST).send('Acces denied');
```
Comment (ike-tom) [inline src/routes/user/verifyToken.js:6]: Access - typo :)

**Review by ike-tom — COMMENTED** [review]
please take a look at my comments

File: src/controllers/user.controllers.js (hunk 4 above)
Comment (PatrykJurczyk) [reply src/controllers/user.controllers.js:42]: TOKEN_SECRET = sekretnytoken

**Review by ike-tom — APPROVED** [review]

**Pull request merged**